src/com/google/common/css/RecordingSubstitutionMap.java

```diff
@@ -68,6 +68,22 @@ public RecordingSubstitutionMap(SubstitutionMap map,
   }
 
   /**
+   * Creates a new instance.
+   * @param map A SubstitutionMap decorated by this class
+   * @param shouldRecordMappingForCodeGeneration A predicate that returns true
+   * @param inputMappings A default one-to-one string mapping
+   *     if the mapping should be recorded for code-generation purposes
+   */
+  public RecordingSubstitutionMap(SubstitutionMap map,
+      Predicate<? super String> shouldRecordMappingForCodeGeneration, Map<String, String> inputMappings) {
+    this(map, shouldRecordMappingForCodeGeneration);
+
+    if (inputMappings != null) {
+      this.mappings.putAll(inputMappings);
```

Comment:
I've got to think about this a bit more. You're right that we need to fill the `RecordingSubstitutionMap`'s mappings from the input, but I'm trying to decide whether this is the right place.